bundles/org.pitest.pitclipse.ui/src/org/pitest/pitclipse/ui/utils/PitclipseUiUtils.java

```diff
@@ -0,0 +1,77 @@
+/*******************************************************************************
+ * Copyright 2021 Lorenzo Bettini and contributors
+ *  
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License.  You may obtain a copy
+ * of the License at
+ *  
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ ******************************************************************************/
+package org.pitest.pitclipse.ui.utils;
+
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.SWTError;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.MessageBox;
+
+/**
+ * A few utilities for the UI.
+ * 
+ * @author Lorenzo Bettini
+ *
+ */
+public class PitclipseUiUtils {
+
+    private PitclipseUiUtils() {
+        // Only static utility methods
+    }
+
+    /**
+     * Only performs the operation if the {@link Composite} is not null and not
+     * disposed.
+     * 
+     * @param c
+     */
+    public static void safeSetFocus(Composite c) {
+        if (c != null && !c.isDisposed()) {
+            c.setFocus();
+        }
+    }
+
+    /**
+     * Only performs the operation if the {@link Composite} is not null and not
+     * disposed.
+     * 
+     * @param c
+     */
+    public static void safeDispose(Composite c) {
+        if (c != null && !c.isDisposed()) {
+            c.dispose();
+        }
+    }
+
+    /**
+     * Executes the {@link Runnable} and in case of {@link SWTError} shows a dialog
+     * with the specified error message, using the specified parent..
+     * 
+     * @param runnable
+     * @param parent
+     * @param errorMessage
+     */
+    public static void safeExecute(Runnable runnable, Composite parent, String errorMessage) {
+        try {
+            runnable.run();
+        } catch (SWTError e) {
+            MessageBox messageBox = new MessageBox(parent.getShell(), SWT.ICON_ERROR | SWT.OK);
+            messageBox.setMessage(errorMessage);
```

Comment:
Shouldn't we also display details about the error? Not sure about the stacktrace but I think that `e`'s message could help the user to fix/understand the issue.

Reply:
But, as you said here https://github.com/pitest/pitclipse/pull/150#discussion_r680315622 , the `SWTError` in such a case contains no message, doesn't it?

Reply:
It's the code snippet that made you think so? I apologize if my comment was misleading, browsing the sources from the IDE seems to indicate that `Browser` does set an error message:

```java
String errMsg = " because there is no underlying browser available.\n";
switch (SWT.getPlatform()) {
    case "gtk":
        errMsg = errMsg + "Please ensure that WebKit with its GTK 3.x bindings is installed (WebKit2 API level is preferred)."
                + " Additionally, please note that GTK4 does not currently have Browser support.\n";
        break;
    case "cocoa":
        errMsg = errMsg + "SWT failed to load the WebKit library.\n";
        break;
    case "win32":
        errMsg = errMsg + "SWT uses either IE or WebKit. Either the SWT.WEBKIT flag is passed and the WebKit library was not "
                + "loaded properly by SWT, or SWT failed to load IE.\n";
        break;
    default:
        break;
}
SWT.error (SWT.ERROR_NO_HANDLES, null, errMsg);
```
Even if there was no message, in the future this function may execute other runnables which may throw a `SWTError` that holds a message. Maybe we can check whether a message is available? Something like:
```java
} catch (SWTError e) {
    if (e.getMessage() != null) {
        errorMessage += "\n" + e.getMessage();
    }
    MessageBox messageBox = ...
    messageBox.setMessage(errorMessage);
}

Reply:
OK, I pushed other commits to address your comments.

In particular, for this comment, I simulated an error like that

```java
    @Override
    public synchronized void createPartControl(Composite parent) {
        PitclipseUiUtils.executeSafely
            (() -> {
                browser = new Browser(parent, SWT.NONE);
                browser.addProgressListener(new PitUiUpdatePublisher(browser));
                throw new SWTError("Example\nAnother line");
            },
            parent,
            "Failed to create the Pit Summary view: browser cannot be initialized.");
    }
```

and I get this message dialog that looks OK, doesn't it?

![image](https://user-images.githubusercontent.com/1202254/127897653-b73f3b4a-8a1f-4c17-a082-b12c08b98cb6.png)

Reply:
Looks good, but is "Exit" a relevant title?

Reply:
It was actually there in the original code, but I agree it sounds strange... any suggestion?

Reply:
I think I'd create a `title` parameter and set the argument to "Failed to create the Pit Summary view":

```java
PitclipseUiUtils.executeSafely
    (() -> {
        ...
    },
    parent,
    "Failed to create the Pit Summary view",  // popup's title
    "Browser cannot be initialized");          // popup's body message
```
If that looks too cumbersome I'd replace "Exit" with "An error occurred", or some similar generic message.

Reply:
Perfect! Here's the result (I've already pushed the additional commit)

![image](https://user-images.githubusercontent.com/1202254/127996740-653010c6-ad13-4b3d-8202-7682743dd84b.png)